src/board.js

```diff
@@ -9,22 +9,232 @@
+    this.turnsSinceLastMove = 0;
+    // Change in number of own/enemy liberties from making a move here.
+    this.ownLibertiesChange = 0;
+    this.enemyLibertiesChange = 0;
+    this.capturesFromMove = 0;  // Number of enemy stones it would capture.
+    this.selfAtariFromMove = 0;  // Number of own stones it would capture.
+    this.sensibleMove = true;  // ie, legal and does not fill its own eyes.
+    this.leadsToLadderCapture = false;
+    this.leadsToLadderEscape = false;
+  }
+
+  Point.prototype = {
+    toString: function() {
+      return rowNameFromNum(this.x) + rowNameFromNum(this.y);
+    },
+  };
+
+  function Group(board, points) {
+    this.board = board;
+    this.color = Board.EMPTY;
+    this.points = new Set();
+    this.liberties = new Set();  // Set of points.
+    points.forEach(this.addPoint.bind(this));
+  }
+
+  Group.prototype = {
+    addPoint: function(point) {
+      var self = this;
+      self.points.add(point);
+      self.color = point.color;
+      point.group = self;
+      self.board.surrounding(point.x, point.y).forEach(function(neighbor) {
+        if (neighbor !== undefined && neighbor.color === Board.EMPTY) {
+          self.liberties.add(neighbor);
+        }
+      });
+    },
+    toString: function() {
+      var points = [];
+      this.points.forEach(function(point) { points.push(point.toString()); });
+      var liberties = [];
+      this.liberties.forEach(function(point) { liberties.push(point.toString()); });
+      return "(" + stringFromColor(this.color) + " group on points " +
+        points.join(", ") + " with liberties " + liberties.join(", ") + ")";
+    },
+  };
+
   // options:
   //   - size: typically 19.
   //   - komi: floating-point number.
   function Board(options) {
     options = options | {};
     this.size = +options.size || 19;
     this.komi = +options.komi || 7.5;
-    this.board = new Uint8Array(this.size * this.size);
+    var numPoints = this.size * this.size;
+    this.board = new Array(numPoints);
+    for (var y = 0; y < this.size; y++) {
+      for (var x = 0; x < this.size; x++) {
+        this.board[x + y * this.size] = new Point(x, y);
+      }
+    }
+    this.groups = new Set();
+    this.nextPlayingColor = Board.BLACK;
   }
 
   Board.prototype = {
     // x, y: integer positions of an intersection on the board.
-    get(x, y) { return this.board[x + y * this.size]; },
-    set(x, y, color) { return this.board[x + y * this.size] = color; },
+    get: function(x, y) { return this.board[x + y * this.size]; },
+    set: function(x, y, color) { this.board[x + y * this.size].color = color; },
+
+    mergeGroups: function(groups) {
+      var points = [];
+      groups.forEach(function(group) {
+        group.points.forEach(function(point) {
+          points.push(point);
+        });
+      });
+      return new Group(this, points);
+    },
+
+    surrounding: function(x, y) {
+      var top    = (y > 0)?             this.get(x    , y - 1): undefined;
```

Comment:
I feel it could be useful, even though it is not currently in use. It does carry the meaning that we are on a given edge, which we wouldn't know otherwise.

But given that we currently don't use that fact, I followed your advice. We can change it back if we ultimately need it.